feat(boom): added boom to orbos

stebenz opened the pull request.

stebenz committed:
fix(customnames): changeable name of cluster and admin-account with the given clusterID

stebenz committed:
docs(examples): added examples for virtualbox

stebenz committed:
fix(error): corrected used id for error in takeoff

stebenz committed:
feat(boom): added code base from boom

stebenz committed:
fix(orbos): combined pipeline for orbiter and boom

stebenz committed:
fix(orbos): define unique names for github actions jobs

stebenz committed:
fix(orbos): unique names and ids for jobs and steps

stebenz committed:
fix(orbos): separated release and dev-release workflows

stebenz committed:
fix(orbos): seperate takeoff for boom, no boom reconciling by orbiter and return when local part of orbiter is finished

stebenz committed:
fix(orbos): combine images and pipelines to orbos

stebenz committed:
fix(orbos): pipeline to result in orbos not orbiter image

stebenz committed:
fix(pipeline): use different folder for charts than artifacts

stebenz committed:
fix(pipeline): moved charts-fetching to dockerbuild

stebenz committed:
fix(pipeline): corrected env-variable for dev-release

stebenz committed:
fix(pipeline): corrected env-variable for delete of old dev-release

stebenz committed:
fix(pipeline): corrected used image in orbiter runtime

stebenz committed:
Merge remote-tracking branch 'origin/efficiency' into customnames

stebenz committed:
perf(merge): merge master

stebenz committed:
fix(boom): corrected deployment of boom into all orbiter clusters

stebenz committed:
fix(boom): corrected start of metrics endpoint

stebenz committed:
fix(orbiter): corrected start of metrics endpoint

stebenz committed:
fix(orbos): moved start of metrics endpoints

stebenz committed:
fix(orbos): integrate fixes of boom repository while merging

stebenz committed:
fix(orbos): fix for chart-fetching

stebenz committed:
fix(orbos): fix to add externalid again after merge

stebenz committed:
fix(orbos): fix marshal when writing secret in boom.yml

stebenz committed:
docs(boom): corrected docs for secret handling

stebenz committed:
fix(boom): corrected handling for repository credentials

stebenz committed:
docs(boom): included boom README.md

stebenz committed:
fix(boom): added option to orbctl to takeoff boom without orbiter

stebenz committed:
fix(boom): only deploy boom when boom.yml is present

stebenz committed:
fix(boom): only deploy boom when boom.yml is present

stebenz committed:
fix(boom): corrected write of array elements with orbctl writesecret

stebenz committed:
fix(boom): corrected read of array elements with orbctl readsecret

stebenz committed:
fix(boom): no panic when grafana-cloud secret is not available

stebenz committed:
docs(example): corrected image in virtualbox example

eliobischof committed:
fix: make ports changeable

stebenz committed:
Merge branch 'boom' into edit-ports

stebenz committed:
fix: make ports changeable

stebenz committed:
fix(boom): only deploy systemd-exporter when deploy-attribute is equal to true

stebenz committed:
fix(boom): self-reconciling with boom related resources

stebenz committed:
fix(boom): self-reconciling with boom related resources

stebenz committed:
fix(boom): fixed import in takeoff

stebenz committed:
fix(boom): fixed no client without kubeconfig

stebenz committed:
fix(boom): added version to deploy-log for reconciling

stebenz committed:
docs(boom): added version for reconciling

stebenz committed:
fix(boom): changed resource names and added labels

eliobischof commented on README.md:
```diff
@@ -15,7 +15,7 @@
-### BOOM
+### [BOOM](docs/boom/boom.md)
 
 > We are currently integrating BOOM into ORBOS see our [Progress](https://github.com/caos/orbos/milestone/1)
```
```suggestion
```

eliobischof commented on cmd/orbctl/takeoff.go:
```diff
@@ -46,98 +50,152 @@ func TakeoffCommand(rv RootValues) *cobra.Command {
+				if err := deployBoom(monitor, gitClient, &value); err != nil {
+					return err
+				}
+			}
```
I think it would be better separation of concerns if start.Orbiter returned the kubeconfig as byte slice?

stebenz replied:
i would even go i step further and build up a list uf kubeconfigs where boom will then get deployed

eliobischof replied:
@stebenz why did you mark this one as resolved?

stebenz replied:
as this code is outdated as per the marking and start.Orbiter() returns a list of kubeconfigs which get used for the deployment of boom

eliobischof commented on internal/git/git.go:
```diff
@@ -100,6 +106,18 @@ func (g *Client) clone() error {
+	}
+	return false
+}
+
```
I would move this function to the package where it is used, because the cloning is a bit counter intuitive and the function is not that generic and only public methods are used

stebenz replied:
I'll move the function to the helper package as it's just for convenience

eliobischof commented on internal/git/git.go:
```diff
@@ -124,12 +142,47 @@ func (g *Client) Read(path string) []byte {
 func (g *Client) StageAndCommit(msg string, files ...File) (bool, error) {
+	if g.stage(files...) {
+		return false, nil
+	}
```
```suggestion
```

eliobischof commented on internal/operator/orbiter/destroy.go:
```diff
@@ -17,9 +17,8 @@ func DestroyFuncGoroutine(query func() error) error {
-func Destroy(monitor mntr.Monitor, gitClient *git.Client, adapt AdaptFunc) error {
-
-	trees, err := parse(gitClient, "orbiter.yml")
+func Destroy(monitor mntr.Monitor, gitClient *git.Client, adapt AdaptFunc, finishedChan chan bool) error {
```
inspect

eliobischof replied:
i think we should replace the finishedChan by context.Context

eliobischof commented on internal/operator/orbiter/kinds/loadbalancers/dynamic/adapt.go:
```diff
@@ -35,7 +35,7 @@ func init() {
 
 func AdaptFunc(whitelist WhiteListFunc) orbiter.AdaptFunc {
-	return func(monitor mntr.Monitor, desiredTree *tree.Tree, currentTree *tree.Tree) (queryFunc orbiter.QueryFunc, destroyFunc orbiter.DestroyFunc, migrate bool, err error) {
+	return func(monitor mntr.Monitor, finishedChan chan bool, desiredTree *tree.Tree, currentTree *tree.Tree) (queryFunc orbiter.QueryFunc, destroyFunc orbiter.DestroyFunc, migrate bool, err error) {
```
should we replace the finishedChan by context.Context?

stebenz replied:
we can replace it, but then it has to be replaced in all adaptFuncs, so i wouldn't change it

eliobischof replied:
yes, that's what i meant. i created the follow-up issue #179

stebenz committed:
fix(boom): corrected usage of pre and post apply folders

stebenz committed:
refactor(orbos): changes after review

stebenz committed:
docs(orbos): update README.md as this is the merge

Co-authored-by: Elio Bischof <eliobischof@gmail.com>

stebenz committed:
docs(boom): corrected documentation for grafana admin

stebenz committed:
fix(orbctl): reload orbconfig in each iteration and if no deploykey is available try oauth-login

stebenz committed:
fix(orbctl): cache github token for deploy key usage

stebenz committed:
feat(config): added config command to rotate masterkey and change the repository

stebenz committed:
fix(build): corrected case-sensitive variable names

stebenz committed:
Merge branch 'orbconfig' into customnames

stebenz committed:
fix(orbiter): custom names for the kubernetes cluster in kubeconfig

stebenz committed:
fix(error): corrected used id for error in takeoff

eliobischof commented:
Writing StaticProvider secret deletes loadbalancing configuration
relates to #181

stebenz commented:
Bug with deleting loadbalancing configuration is corrected in #185

stebenz committed:
Merge branch 'boom' into boomreconciling

stebenz committed:
fix(merge): correction after merge

stebenz committed:
feat(boom): reconciling of boom version

feat(boom): reconciling of boom version

stebenz committed:
Merge branch 'boom' into orbconfig

stebenz committed:
feat(orbctl): added config command and several additional features to takeoff and teardown

feat(orbctl): added config command and several additional features to takeoff and teardown

eliobischof commented:
![image](https://user-images.githubusercontent.com/12727842/83763578-275ecf80-a679-11ea-9b28-52e4b5208514.png)

I think the config command should log what it has done or not done

eliobischof commented on internal/operator/orbiter/kinds/clusters/kubernetes/client.go:
```diff
@@ -49,7 +50,12 @@ type Client struct {
+		if err != nil {
+			monitor.Error(err)
+		}
+	}
```
```suggestion
		err := kc.Refresh(kubeconfig)
		if err != nil {
		    // do nothing
		}
```

eliobischof replied:
![image](https://user-images.githubusercontent.com/12727842/83764557-5cb7ed00-a67a-11ea-995c-1d0fc007139b.png)

eliobischof replied:
when creating a new client without having a kubeconfig, it's not an error. This behaviour is convenient because of the bootstrapping phase when no cluster exists already

stebenz committed:
fix(orbctl): Added logs for config-command

eliobischof commented on internal/operator/boom/application/applications/prometheus/helm.go:
```diff
@@ -0,0 +1,139 @@
+	ingestionSecretAbsent := k8serrors.IsNotFound(errors.Unwrap(getSecretErr))
+	if getSecretErr != nil && !ingestionSecretAbsent {
+		// TODO: Better error handling?
+		monitor.Error(getSecretErr)
```
```suggestion
		monitor.Info("Not sending telemetry data to MISSION as secret grafana-cloud is missing in namespace caos-system")
```

eliobischof committed:
Update debug.sh

eliobischof committed:
fix: disable CGO

stebenz committed:
fix(client): no kubeconfig is no error

Co-authored-by: Elio Bischof <eliobischof@gmail.com>

stebenz committed:
refactor(error): better error message when secret not existing

Co-authored-by: Elio Bischof <eliobischof@gmail.com>

eliobischof commented on scripts/debug.sh:
```diff
@@ -2,5 +2,6 @@
\ No newline at end of file
+go run ./cmd/gen-executables/*.go --debug --version $(git rev-parse --abbrev-ref HEAD | sed -e "s/heads\///") --commit $(git rev-parse HEAD) --orbctl ./artifacts 
+CGO_ENABLED=0 GOOS=linux go build -o ./artifacts/gen-charts  cmd/gen-charts/*.go
+dlv exec --api-version 2 --headless --listen 127.0.0.1:5000 /tmp/orbctldev/orbctl-Linux-x86_64 -- $@
```
```suggestion
dlv exec --api-version 2 --headless --listen 127.0.0.1:5000 ./artifacts/orbctl-Linux-x86_64 -- $@
```

eliobischof committed:
Update scripts/debug.sh

eliobischof commented on internal/operator/orbiter/kinds/clusters/kubernetes/adapt.go:
```diff
@@ -95,8 +116,12 @@ func AdaptFunc(
 			Current: current,
 		}
 
+		finishedChan <- finished
```
```suggestion
                go func(){
		    finishedChan <- finished
		}()
```

eliobischof committed:
Update internal/operator/orbiter/kinds/clusters/kubernetes/adapt.go

eliobischof commented:
If cloning doesn't work, we should throw it as error
![image](https://user-images.githubusercontent.com/12727842/83875570-a1ee2480-a737-11ea-84f9-f421b4e379d6.png)

eliobischof commented:
![image](https://user-images.githubusercontent.com/12727842/83878294-38244980-a73c-11ea-839b-d3cbb66bfaeb.png)

stebenz committed:
fix(orbctl): fixed writeback as the local orbconfig has the be written when the repoURL changed as well

eliobischof commented:
Now, the local orbconfig is updated but not the in-cluster secret 
![image](https://user-images.githubusercontent.com/12727842/83879864-9f42fd80-a73e-11ea-9e19-b9f7bb211804.png)

stebenz committed:
fix(orbctl): fixed config so that kubeconfigs will get read out of orbiter.yml
